specification/web/resource-manager/Microsoft.DomainRegistration/stable/2024-11-01/Domains.json

```diff
@@ -0,0 +1,1616 @@
+            },
+            "createdTime": {
+              "format": "date-time",
+              "description": "Domain creation timestamp.",
+              "type": "string",
+              "readOnly": true
+            },
+            "expirationTime": {
+              "format": "date-time",
+              "description": "Domain expiration timestamp.",
+              "type": "string",
+              "readOnly": true
+            },
+            "lastRenewedTime": {
+              "format": "date-time",
+              "description": "Timestamp when the domain was renewed last time.",
+              "type": "string",
+              "readOnly": true
+            },
+            "autoRenew": {
+              "description": "<code>true</code> if the domain should be automatically renewed; otherwise, <code>false</code>.",
+              "default": true,
+              "type": "boolean"
+            },
+            "readyForDnsRecordManagement": {
+              "description": "<code>true</code> if Azure can assign this domain to App Service apps; otherwise, <code>false</code>. This value will be <code>true</code> if domain registration status is active and \n it is hosted on name servers Azure has programmatic access to.",
+              "type": "boolean",
+              "readOnly": true
+            },
+            "managedHostNames": {
+              "description": "All hostnames derived from the domain and assigned to Azure resources.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/HostName"
+              },
+              "readOnly": true,
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "consent": {
+              "$ref": "#/definitions/DomainPurchaseConsent",
+              "description": "Legal agreement consent.",
+              "x-ms-mutability": [
+                "create"
+              ]
+            },
+            "domainNotRenewableReasons": {
+              "description": "Reasons why domain is not renewable.",
+              "type": "array",
+              "items": {
+                "enum": [
+                  "RegistrationStatusNotSupportedForRenewal",
+                  "ExpirationNotInRenewalTimeRange",
+                  "SubscriptionNotActive"
+                ],
+                "type": "string",
+                "x-ms-enum": {
+                  "name": "ResourceNotRenewableReason",
+                  "modelAsString": true
+                }
+              },
+              "readOnly": true
+            },
+            "dnsType": {
+              "description": "Current DNS type",
+              "enum": [
+                "AzureDns",
+                "DefaultDomainRegistrarDns"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DnsType",
+                "modelAsString": false
+              }
+            },
+            "dnsZoneId": {
```

Comment:
>dnsZoneId [](http://example.com/codeflow?start=13&length=9)

If this is an Azure resource ID, recommend dnsZoneResourceId. Otherwise, add details in the description (e.g. Guid, integer, name, etc.).